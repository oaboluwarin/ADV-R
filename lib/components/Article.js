import React, { PureComponent } from 'react';
import { object, shape, string } from 'prop-types';
import { styles } from '..';
import { withStore } from './withStore';

const dateDisplay = (dateString) => new Date(dateString).toDateString();

class Article extends PureComponent {
  static propTypes = {
    article: shape({
      date: string.isRequired,
      title: string.isRequired,
      body: string.isRequired,
    }),
    store: object
  }

  render() {
    const { article, author } = this.props;

    return (
      <div style={styles.article}>
        <div style={styles.title}>{article.title}</div>
        <div style={styles.date}>
          {dateDisplay(article.date)}
        </div>
        <div style={styles.author}>
          <a href={author.website}>
            {author.firstName} {author.lastName}
          </a>
        </div>
        <div style={styles.body}>{article.body}</div>
      </div>
    );
  }
}

const extraProps = (store, originalProps) => {
  return {
    author: store.lookUpAuthor(originalProps.article.authorId),
  };
};

export default withStore(extraProps)(Article);
