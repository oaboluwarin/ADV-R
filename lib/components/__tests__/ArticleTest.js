import React from 'react';
import renderer from 'react-test-renderer';
import Article from '../Article';

describe('Article', () => {

  const testProps = {
    article: {
      title: 'Winds of winter',
      date: 'Sun Mar 18 2018 00:00:00 GMT+0000 (UTC)',
      body: 'The night king embarks on a quest to wipe out all of humanity'
    },
    store: {
      lookUpAuthor: jest.fn(() => ({
        author: {
          website: 'http://wordifice.wordpress.com',
          firstName: 'oreoluwa',
          lastName: 'aboluwarin'
        }
      })),
    }
  };

  it('renders the Article component correctly', () => {
    const tree = renderer.create(
      <Article
        {...testProps}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
