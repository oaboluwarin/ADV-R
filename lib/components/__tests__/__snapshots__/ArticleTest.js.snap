// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders the Article component correctly 1`] = `
<div
  style={
    Object {
      "borderBottomColor": "#aaa",
      "borderBottomStyle": "solid",
      "borderBottomWidth": 1,
      "marginBottom": 10,
      "paddingBottom": 10,
    }
  }
>
  <div
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Winds of winter
  </div>
  <div
    style={
      Object {
        "color": "#888",
        "fontSize": "0.85em",
      }
    }
  >
    Sun Mar 18 2018
  </div>
  <div
    style={
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <a
      href={undefined}
    >
       
    </a>
  </div>
  <div
    style={
      Object {
        "paddingLeft": 20,
      }
    }
  >
    The night king embarks on a quest to wipe out all of humanity
  </div>
</div>
`;
